### PR TITLE
Add angle and destinationOnSegment

### PR DESCRIPTION
**Pull request by trandaison**
- Add 2 new functions:
  - `angle`: Returns the the angle of the given segment and the meridian in degrees, clockwise from 0 degrees east. 
  - `destinationOnSegment`:  Returns the point that is a distance and heading away from the given origin point and snap on the given segment.
- Update the docs.

Please review and merge this one, thank you!

### CODE REVIEW

**Comment by mdartic** [comment]
Thank you @trandaison !
Will you agree adding some test case for your two new methods ?
If edge cases exist, please integrate them too ?

File: src/leaflet.geometryutil.js [hunk 1]
```diff
@@ -703,8 +703,8 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
        Returns the point that is a distance and heading away from
        the given origin point.
        @param {L.LatLng} latlng: origin point
-       @param {float}: heading in degrees, clockwise from 0 degrees north.
-       @param {float}: distance in meters
+       @param {float} heading: heading in degrees, clockwise from 0 degrees north.
+       @param {float} distance: distance in meters
```
Comment (leplatrem) [inline src/leaflet.geometryutil.js]: 👍

File: src/leaflet.geometryutil.js [hunk 2]
```diff
@@ -728,7 +728,38 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
         lon2 = lon2 * radInv;
         lon2 = lon2 > 180 ? lon2 - 360 : lon2 < -180 ? lon2 + 360 : lon2;
         return L.latLng([lat2 * radInv, lon2]);
-    }
+    },
+
+    /**
+       Returns the the angle of the given segment and the meridian in degrees,
+       clockwise from 0 degrees north.
+       @param {L.Map} map Leaflet map to be used for this method
```
Comment (leplatrem) [inline src/leaflet.geometryutil.js]: nit: `map:`

File: src/leaflet.geometryutil.js [hunk 3]
```diff
@@ -728,7 +728,38 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
...
+       @param {L.LatLng} latlngA: geographical point A of the segment
+       @param {L.LatLng} latlngB: geographical point B of the segment
+       @returns {Float} the angle in degrees.
+    */
+    angle: function(map, latlngA, latlngB) {
+      var pointA = map.latLngToContainerPoint(latlngA),
+          pointB = map.latLngToContainerPoint(latlngB),
+          angleDeg = Math.atan2(pointB.y - pointA.y, pointB.x - pointA.x) * 180 / Math.PI + 90;
+      angleDeg += angleDeg < 0 ? 360 : 0;
+      return angleDeg;
```
Comment (leplatrem) [inline src/leaflet.geometryutil.js]: Ideas of tests: vertical and horizontal segments, negative angle

File: src/leaflet.geometryutil.js [hunk 4]
```diff
@@ -728,7 +728,38 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
...
+    },
+
+    /**
+       Returns the point that is a distance and heading away from
+       the given origin point and snap on the given segment.
+       @param {L.Map} map Leaflet map to be used for this method
```
Comment (leplatrem) [inline src/leaflet.geometryutil.js]: nit: `map:`

File: src/leaflet.geometryutil.js [hunk 5]
```diff
@@ -728,7 +728,38 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
...
+    angle: function(map, latlngA, latlngB) {
+      var pointA = map.latLngToContainerPoint(latlngA),
+          pointB = map.latLngToContainerPoint(latlngB),
+          angleDeg = Math.atan2(pointB.y - pointA.y, pointB.x - pointA.x) * 180 / Math.PI + 90;
+      angleDeg += angleDeg < 0 ? 360 : 0;
+      return angleDeg;
+    },
+
+    /**
+       Returns the point that is a distance and heading away from
```
Comment (leplatrem) [inline src/leaflet.geometryutil.js]: *is at a distance* ?

**Comment by trandaison** [comment]
@mdartic
Thanks for your comment.
If edge cases exist, it would be anticlockwise.This function return an angle in clockwise from 0 degrees north. In case you want to get the edge angle, `360 - anggle` would do the job.

**Comment by mdartic** [comment]
Thanks @trandaison !
